langchain/src/chat_models/base.ts

```diff
@@ -0,0 +1,864 @@
+  ): ReturnType<BaseChatModel["withStructuredOutput"]> => {
+    this._queuedMethodOperations.withStructuredOutput = [schema, ...args];
+    return new _ConfigurableModel<RunInput, CallOptions>({
+      defaultConfig: this._defaultConfig,
+      configurableFields: this._configurableFields,
+      configPrefix: this._configPrefix,
+      queuedMethodOperations: this._queuedMethodOperations,
+    }) as unknown as ReturnType<BaseChatModel["withStructuredOutput"]>;
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _modelParams(config?: RunnableConfig): Record<string, any> {
+    const configurable = config?.configurable ?? {};
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let modelParams: Record<string, any> = {};
+
+    for (const [key, value] of Object.entries(configurable)) {
+      if (key.startsWith(this._configPrefix)) {
+        const strippedKey = this._removePrefix(key, this._configPrefix);
+        modelParams[strippedKey] = value;
+      }
+    }
+
+    if (this._configurableFields !== "any") {
+      modelParams = Object.fromEntries(
+        Object.entries(modelParams).filter(([key]) =>
+          this._configurableFields.includes(key)
+        )
+      );
+    }
+
+    return modelParams;
+  }
+
+  _removePrefix(str: string, prefix: string): string {
+    return str.startsWith(prefix) ? str.slice(prefix.length) : str;
+  }
+
+  /**
+   * Bind config to a Runnable, returning a new Runnable.
+   * @param {RunnableConfig | undefined} [config] - The config to bind.
+   * @returns {RunnableBinding<RunInput, RunOutput, CallOptions>} A new RunnableBinding with the bound config.
+   */
+  withConfig(
+    config?: RunnableConfig
+  ): RunnableBinding<RunInput, AIMessageChunk, CallOptions> {
+    const mergedConfig: RunnableConfig = { ...(config || {}) };
+    const modelParams = this._modelParams(mergedConfig);
+
+    const remainingConfig: RunnableConfig = Object.fromEntries(
+      Object.entries(mergedConfig).filter(([k]) => k !== "configurable")
+    );
+
+    remainingConfig.configurable = Object.fromEntries(
+      Object.entries(mergedConfig.configurable || {}).filter(
+        ([k]) =>
+          this._configPrefix &&
+          !Object.keys(modelParams).includes(
+            this._removePrefix(k, this._configPrefix)
+          )
+      )
+    );
+
+    const newConfigurableModel = new _ConfigurableModel<RunInput, CallOptions>({
+      defaultConfig: { ...this._defaultConfig, ...modelParams },
+      configurableFields: Array.isArray(this._configurableFields)
+        ? [...this._configurableFields]
+        : this._configurableFields,
+      configPrefix: this._configPrefix,
+    });
+
+    return new RunnableBinding<RunInput, AIMessageChunk, CallOptions>({
+      config: mergedConfig,
+      bound: newConfigurableModel,
+    });
+  }
+
+  async invoke(
+    input: RunInput,
+    options?: CallOptions
+  ): Promise<AIMessageChunk> {
+    const model = await this._model(options);
```

Comment:
nit: why are we reloading this every time?

Reply:
Runtime call options require it